File: ruby/reverse_string.rb


def reverse_string(str)
  # type your code in here
  reverse_string = ""
  str.length.times do |i|
    reverse_string = str[i] + reverse_string
  end
  reverse_string
  
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file

# take the string and turn into array
# take the last element of array and put it first
# print to console 



# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  let reverse_string = ""
  for (let i = 0; i < str.length; i++){
    reverse_string = str[i] + reverse_string
  }
  return reverse_string
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
